packages/machine-extractor/src/scalars.ts

```diff
@@ -4,87 +4,88 @@ import {
   maybeIdentifierTo,
   memberExpressionReferencingEnumMember,
 } from "./identifiers";
-import { StringLiteralNode } from "./types";
 import { maybeTsAsExpression } from "./tsAsExpression";
+import { StringLiteralNode } from "./types";
 import { unionType } from "./unionType";
 import { wrapParserResult } from "./wrapParserResult";
 
 export const StringLiteral = unionType([
-  wrapParserResult(memberExpressionReferencingEnumMember, (node) => {
+  wrapParserResult(memberExpressionReferencingEnumMember, ({ path, value }) => {
     return {
-      node: node.node as t.Node,
-      value: node.value,
+      path,
+      node: path.node as t.Node,
+      value: value,
     };
   }),
   maybeTsAsExpression(
     maybeIdentifierTo(
       createParser({
         babelMatcher: t.isStringLiteral,
-        parseNode: (node): StringLiteralNode => {
+        extract: (path): StringLiteralNode => {
           return {
-            value: node.value,
-            node,
+            value: path.node.value,
+            node: path.node,
+            path,
           };
         },
-      }),
-    ),
+      })
+    )
   ),
 ]);
 
 export const NumericLiteral = maybeTsAsExpression(
   maybeIdentifierTo(
     createParser({
       babelMatcher: t.isNumericLiteral,
-      parseNode: (node) => {
+      extract: (path) => {
         return {
-          value: node.value,
-          node,
+          value: path.node.value,
+          node: path.node,
+          path,
         };
       },
-    }),
-  ),
+    })
+  )
 );
 
 export const BooleanLiteral = maybeTsAsExpression(
   maybeIdentifierTo(
     createParser({
       babelMatcher: t.isBooleanLiteral,
-      parseNode: (node) => {
+      extract: (path) => {
         return {
-          value: node.value,
-          node,
+          value: path.node.value,
+          node: path.node,
+          path,
         };
       },
-    }),
-  ),
+    })
+  )
 );
 
 export const AnyNode = createParser({
   babelMatcher: t.isNode,
-  parseNode: (node) => ({ node }),
+  extract: (path) => ({ path, node: path.node }),
 });
 
 export const Identifier = createParser({
   babelMatcher: t.isIdentifier,
-  parseNode: (node) => ({ node }),
+  extract: (path) => ({ path, node: path.node }),
 });
 
 export const TemplateLiteral = maybeTsAsExpression(
   maybeIdentifierTo(
     createParser({
       babelMatcher: t.isTemplateLiteral,
-      parseNode: (node) => {
-        let value = "";
+      extract: (path) => {
+        const evaluated = path.evaluate();
```

Comment:
We might need to do some additional work here since `evaluate` might not work correctly if it encounters unidentifiable `quasis`. Maybe we just return `null` if that's the case because it's likely an issue on the user's end?